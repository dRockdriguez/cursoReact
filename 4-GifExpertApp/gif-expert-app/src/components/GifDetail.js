import React from "react";
import PropTypes from 'prop-types';

const GifDetail = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title} </p>
    </div>
  );
};

GifDetail.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};
export default GifDetail;
