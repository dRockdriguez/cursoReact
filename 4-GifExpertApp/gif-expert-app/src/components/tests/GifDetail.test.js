import React from 'react';
import GifDetail from '../GifDetail';
import { shallow } from 'enzyme';

describe('GifDetail', () => {
    const title = 'TÃ­tulo';
    const url = 'url';
    const wrapper = shallow(<GifDetail title={ title } url={ url } />);

    test('Should render the component', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('Should render the title in p', () => {
        const p = wrapper.find('p').text().trim();
        expect(p).toBe(title);
    })

    test('Should render the img with src and alt', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('Should render the div with card class', () => {
        const div = wrapper.find('div');

        expect(div.prop('className').includes('card')).toBeTruthy();
    })
});