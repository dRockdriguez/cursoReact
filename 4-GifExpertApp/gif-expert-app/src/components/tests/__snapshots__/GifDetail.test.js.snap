// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifDetail Should render the component 1`] = `
<div
  className="card"
>
  <img
    alt="Título"
    src="url"
  />
  <p>
    Título
     
  </p>
</div>
`;
